tests/test_paragraph_hash.py

```
@@ -0,0 +1,418 @@
+
+class TestScopedOperations:
+    @pytest.fixture
+    def temp_docx(self):
+        test_data = Path(__file__).parent / "test_data" / "simple.docx"
+        temp = tempfile.mkdtemp(prefix="docx_hash_test_")
+        dest = Path(temp) / "test.docx"
+        shutil.copy(test_data, dest)
+        yield dest
+        shutil.rmtree(temp, ignore_errors=True)
+
+    def _get_paragraph_ref(self, doc, index):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_get_paragraph_ref`

(ANN202)